### PR TITLE
Up-to-date build script for Summit

### LINKED ISSUE #524: Outdated summit script
When I source scripts/summitVariables I am getting the following errors:
 source scripts/summitVariables.sh 
`Lmod has detected the following error:  The following module(s) are unknown: "gcc/7.4.0"`
```
Lmod has detected the following error:  These module(s) or extension(s) exist but cannot be loaded as requested: "m4/1.4.18"
   Try: "module spider m4/1.4.18" to see how to load the module(s).
Lmod has detected the following error:  The following module(s) are unknown: "python/3.7.0"
```

Lmod has detected the following error:  The following module(s) are unknown: "openmpi/4.0.3"

```
Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore_cache load "openmpi/4.0.3"
```

### PR DESCRIPTION
**Pull request by nkoukpaizan**
Closes #524

### CODE REVIEW

**Review by CameronRutherford — CHANGES_REQUESTED** [review]
LGTM once my comment is addressed. Can you also include the 'spack.yaml' you used to generate these modules? I want to start adding those to the HiOp repo as well. I'm thinking a subfolder in scripts would work?

File: scripts/summitVariables.sh [hunk 1]
```diff
@@ -1,60 +1,46 @@
...
-module load bison
-module load diffutils
-module load cmake/3.20.2
-module load openmpi/4.0.3
+export CC=/sw/summit/gcc/10.2.0-2/bin/gcc
+export CXX=/sw/summit/gcc/10.2.0-2/bin/g++
+export FC=/sw/summit/gcc/10.2.0-2/bin/gfortran
 
 [ -f $PWD/nvblas.conf ] && rm $PWD/nvblas.conf
 cat > $PWD/nvblas.conf <<-EOD
```
Comment (pelesh) [inline scripts/summitVariables.sh:46]: Do we still need to generate `nvblas.conf`? I am not sure if HiOp is using nvblas anymore.

File: scripts/summitVariables.sh [hunk 2]
```diff
@@ -67,8 +53,15 @@ EOD
...
+EXTRA_CMAKE_ARGS="$EXTRA_CMAKE_ARGS -DMPI_C_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicc"
+EXTRA_CMAKE_ARGS="$EXTRA_CMAKE_ARGS -DMPI_CXX_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpicxx"
+EXTRA_CMAKE_ARGS="$EXTRA_CMAKE_ARGS -DMPI_Fortran_COMPILER:STRING=/sw/summit/spack-envs/base/opt/linux-rhel8-ppc64le/gcc-10.2.0/spectrum-mpi-10.4.0.3-20210112-ht5bw4jruhjujvkzcvpra5lryg5vfhy4/bin/mpif90"
+EXTRA_CMAKE_ARGS="$EXTRA_CMAKE_ARGS -DHIOP_UMFPACK_DIR:STRING=/autofs/nccs-svm1_proj/csc359/shared_installs/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/suite-sparse-5.10.1-bgorozsknorc3meyrd3qtg3yzxwlcv2q"
+EXTRA_CMAKE_ARGS="$EXTRA_CMAKE_ARGS -DHIOP_COINHSL_DIR:STRING=/autofs/nccs-svm1_proj/csc359/shared_installs/spack/opt/spack/linux-rhel8-ppc64le/gcc-10.2.0/coinhsl-2015.06.23-kio55bvmdzjlsqpn63nlywbtlijky6pu"
```
[the remark this answers is not in the document]
Comment (nkoukpaizan) [reply scripts/summitVariables.sh:67]: I modeled the script from the ascentVariables.sh, which had those extra variables. I'll test whether they are needed or not.

File: scripts/summitVariables.sh (hunk 1 above)
Comment (nkoukpaizan) [reply scripts/summitVariables.sh:46]: We might want @cnpetra to confirm, but tests are passing for me without including an `nvblas.conf` file.

File: scripts/platforms/summit/spack.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+spack:
+  specs:
+  - hiop@develop+cuda+raja+sparse+mpi+deepchecking+ginkgo+cusolver cuda_arch=70 build_type=Release
```
Comment (nychiang) [inline scripts/platforms/summit/spack.yaml:3]: @cnpetra Do you want to activate `deepchecking` in the `Release` mode?

Comment (cnpetra) [reply scripts/platforms/summit/spack.yaml:3]: makes no sense

Comment (nychiang) [reply scripts/platforms/summit/spack.yaml:3]: @nkoukpaizan @CameronRutherford let's remove `deepchecking` when `build_type=Release`

Comment (pelesh) [reply scripts/platforms/summit/spack.yaml]: Deep checking was added for builds used on CI pipelines. Somehow that slipped into other builds.

**Review by cnpetra — APPROVED** [review]
lgtm, looks like this is addressing (and closing) Issue#524 open by @kswirydo, let me know if it does not.

File: scripts/platforms/summit/spack.yaml (hunk 3 above)
Comment (CameronRutherford) [reply scripts/platforms/summit/spack.yaml]: We shoud take this into consideration when merging https://github.com/spack/spack/pull/32750 @ryandanehy

**Review by CameronRutherford — APPROVED** [review]

**Pull request merged**